linux_pipeline/Jenkinsfile_azure_new_vm_size_provision

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env groovy
+def RunPowershellCommand(psCmd) {
+    bat "powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \"[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;$psCmd;EXIT \$global:LastExitCode\""
+}
+
+
+env.NewVMsEnv = ""
+env.todaysSizes = ""
+env.current = ""
+env.CurrentSize =  ""
+env.CurrentRegion =  ""
+env.NewVMSizeEmailBody = "SR\t\tSize\t\tRegion\n"
+env.test_cmd= "BVT-VERIFY-DEPLOYMENT-PROVISION"
+stage("Checkout")
+{
+    timeout(120) {
+        node("master")
+        {
+            cleanWs()
+            git "https://github.com/LIS/lis-pipeline.git"
+            stash includes: '**', name: 'lis-pipeline'
+            cleanWs()
+            git "https://github.com/LIS/LISAv2.git"
+            stash includes: '**', name: 'LISAv2'
+        }
+    }
+}
+
+stage ("Check for new sizes")
+{
+    node ("azure")
+    {
+        unstash 'lis-pipeline'
+        withCredentials([file(credentialsId: 'Azure_Secrets_File', variable: 'Azure_Secrets_File')])
+        {
+            RunPowershellCommand(".\\scripts\\azure_new_vm_size_provision\\watcher-new-azure-vm-size.ps1")
+        }
+        NewVMsEnv = readFile 'todaysNewVMs.txt'
+        println(NewVMsEnv)
+    }
+}
+stage ("Execute Tests")
+{
+    node ("azure")
+    {
+        if (NewVMsEnv == 'NO_NEW_VMS')
+        {
+            println ("No new VMs today.")
+        }
+        else
+        {
+            println ("Testing new VMs... ")
+            def Counter = 1
+            todaysSizes = NewVMsEnv.split(",")
+            for( item in todaysSizes)
+            {
+                cleanWs()
+                unstash 'LISAv2'
+                CurrentRegion = item.split(" ")[0]
+                CurrentSize = item.split(" ")[1]
+                NewVMSizeEmailBody += "${Counter}.\t\tCurrentSize\t\tCurrentRegion\n"
+                withCredentials([file(credentialsId: 'Azure_Secrets_File', variable: 'Azure_Secrets_File')])
+                {
+                echo "LisaV2: Runing provision tests in ${CurrentRegion} with VM size ${CurrentSize}..."
+                RunPowershellCommand(".\\Run-LisaV2.ps1" +
+                    " -RGIdentifier 'NewVM'" +
+                    " -TestLocation '${CurrentRegion}'" +
+                    " -TestPlatform 'Azure'" +
+                    " -ARMImageName 'Canonical UbuntuServer 16.04-LTS latest'" +
+                    " -XMLSecretFile '${Azure_Secrets_File}'" +
+                    " -StorageAccount 'ExistingStorage_Standard'" +
+                    " -OverrideVMSize '${CurrentSize}'" +
+                    " -ExitWithZero"+
+                    " ${test_cmd}"
+                )
+                junit "Report\\*-junit.xml"
+                archiveArtifacts "TestResults\\**\\*"
+                }
+            }
+            emailext body: "${NewVMSizeEmailBody}", subject: 'Jenkins CI : New VM size(s) found', to: 'lisval@microsoft.com'
```

Comment:
please store the mail value as a secret, do not hardcode it here

Reply:
Fixed